go-controller/pkg/observability/observability_test.go

```diff
@@ -0,0 +1,135 @@
+package observability
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	libovsdbtest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/libovsdb"
+)
+
+var _ = Describe("Observability Manager", func() {
+	var (
+		nbClient        libovsdbclient.Client
+		libovsdbCleanup *libovsdbtest.Context
+		manager         *Manager
+	)
+
+	const collectorUUID = "collector-uuid"
+
+	startManager := func(data []libovsdbtest.TestData) {
+		var err error
+		nbClient, _, libovsdbCleanup, err = libovsdbtest.NewNBSBTestHarness(libovsdbtest.TestSetup{
+			NBData: data})
+		Expect(err).NotTo(HaveOccurred())
+		manager = NewManager(nbClient)
+		err = manager.Init()
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	createACLWithPortGroup := func(acl *nbdb.ACL) *nbdb.PortGroup {
+		ops, err := libovsdbops.CreateOrUpdateACLsOps(nbClient, nil, manager.SamplingConfig(), acl)
+		Expect(err).NotTo(HaveOccurred())
+		pg := &nbdb.PortGroup{
+			UUID: "pg-uuid",
+			ACLs: []string{acl.UUID},
+		}
+		ops, err = libovsdbops.CreateOrUpdatePortGroupsOps(nbClient, ops, pg)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = libovsdbops.TransactAndCheck(nbClient, ops)
+		Expect(err).NotTo(HaveOccurred())
+		return pg
+	}
+
+	initialDB := []libovsdbtest.TestData{
+		&nbdb.SamplingApp{
+			UUID: "drop-sampling-uuid",
+			ID:   DropSamplingID,
+			Type: nbdb.SamplingAppTypeDrop,
+		},
+		&nbdb.SamplingApp{
+			UUID: "acl-new-traffic-sampling-uuid",
+			ID:   ACLNewTrafficSamplingID,
+			Type: nbdb.SamplingAppTypeACLNew,
+		},
+		&nbdb.SamplingApp{
+			UUID: "acl-est-traffic-sampling-uuid",
+			ID:   ACLEstTrafficSamplingID,
+			Type: nbdb.SamplingAppTypeACLEst,
+		},
+		&nbdb.SampleCollector{
+			UUID:        collectorUUID,
+			ID:          1,
+			SetID:       DefaultObservabilityCollectorSetID,
+			Probability: 65535,
+		},
+	}
+
+	AfterEach(func() {
+		if libovsdbCleanup != nil {
+			libovsdbCleanup.Cleanup()
+		}
+	})
+
+	for _, dbSetup := range [][]libovsdbtest.TestData{
+		nil, initialDB,
+	} {
+		msg := "db is empty"
+		if dbSetup != nil {
+			msg = "db is not empty"
+		}
+		When(msg, func() {
+
+			It("should initialize database", func() {
+				startManager(dbSetup)
+				Eventually(nbClient).Should(libovsdbtest.HaveData(initialDB))
```

Comment:
If we add stale Sample_Collector cleanup we should probably test for that too here somewhere.